rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -171,8 +171,64 @@ def _decode_vote_cast(self, context: DecoderContext) -> DecodingOutput:
 
 class ReloadableDecoderMixin(metaclass=ABCMeta):
 
+    def __init__(
+            self,
+            cache_type_to_check_for_freshness: CacheType,
+            query_data_method: Callable,
+            save_data_to_cache_method: Callable,
+            read_pools_and_gauges_from_cache_method: Callable,
+    ) -> None:
+        """
+        :param cache_type_to_check_for_freshness: The cache type that is checked for freshness.
+        :param query_data_method: The method that queries the remote source for data.
+        :param save_data_to_cache_method: The method that saves the data to the cache tables.
+        :param read_pools_and_gauges_from_cache_method: The method that reads the data from the
+        cache tables.
+        """
+        self.cache_type_to_check_for_freshness = cache_type_to_check_for_freshness
+        self.query_data_method = query_data_method
+        self.save_data_to_cache_method = save_data_to_cache_method
+        self.read_pools_and_gauges_from_cache_method = read_pools_and_gauges_from_cache_method
+        self.pools, self.gauges = self.read_pools_and_gauges_from_cache_method()
+
     @abstractmethod
+    def _decode_pool_events(self, context: DecoderContext) -> DecodingOutput:
+        """Subclasses should implement this method. It decodes pool related events."""
+
+    @abstractmethod
+    def _decode_gauge_events(self, context: DecoderContext) -> DecodingOutput:
+        """Subclasses should implement this method. It decodes gauge related events."""
+
     def reload_data(self) -> Optional[Mapping[ChecksumEvmAddress, tuple[Any, ...]]]:
-        """Subclasses may implement this to be able to reload some of the decoder's properties
-        Returns only new mappings of addresses to decode functions
+        """Make sure pools and gauges are recently queried from the remote source,
+        saved in the DB and loaded to the decoder's memory.
+
+        If a query happens and any new mappings are generated they are returned,
+        otherwise `None` is returned.
         """
+        self.evm_inquirer.ensure_cache_data_is_updated(  # type: ignore  # mypy doesn't understand that the optimism inquirer is a DSProxyInquirerWithCacheData with an ensure_cache_data_is_updated method  # noqa: E501
+            cache_type=self.cache_type_to_check_for_freshness,
```

Comment:
this is not totally correct. This is missing the evm inquirer in the definition of the interface and this is why it fails first. Later it fails because of the dsproxy issue but this is missing the evm_inquierer definition in the interface